# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1265267431 -3600
#      Thu Feb 04 08:10:31 2010 +0100
# Node ID 2b05f4aff27c94bd21c6f02f38ab3c7d0e0809bf
# Parent  a4fb4675accb3708d50a89a6cca979f7a97128d4
fix off-by-one month error in dir

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-02-04  Jaroslav Hajek  <highegg@gmail.com>
+
+	* miscellaneous/dir.m: Fix month passed to datenum.
+
 2010-02-03  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/private/__isequal__.m: Handle function handles.
 
 2010-02-01  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m (get_old_gnuplot_color): Remove as
 	no longer used (for gnuplot 4.0).
diff --git a/scripts/miscellaneous/dir.m b/scripts/miscellaneous/dir.m
--- a/scripts/miscellaneous/dir.m
+++ b/scripts/miscellaneous/dir.m
@@ -112,17 +112,17 @@ function retval = dir (file)
 	  endif
 	  [dummy, fn, ext] = fileparts (fn);
 	  fn = cstrcat (fn, ext);
 	  info(i,1).name = fn;
 	  lt = localtime (st.mtime);
 	  info(i,1).date = strftime ("%d-%b-%Y %T", lt);
 	  info(i,1).bytes = st.size;
 	  info(i,1).isdir = S_ISDIR (st.mode);
-	  info(i,1).datenum = datenum (lt.year + 1900, lt.mon, lt.mday,
+	  info(i,1).datenum = datenum (lt.year + 1900, lt.mon + 1, lt.mday,
 				       lt.hour, lt.min, lt.sec);
 	  info(i,1).statinfo = st;
 	endif
       endfor
     endif
 
   else
     error ("dir: expecting directory or filename to be a char array");
